### DISCUSSION TITLE
SQLAlchemy 1.4

### DISCUSSION THREAD

**Discussion by Timusan** [fastapi-users/fastapi-users#554] [General]
Just a general bump to ping if SQLAlchemy 1.4 (which is released in stable a couple of days ago) is on the radar to be implemented? Newer projects are highly recommended to use the (improved) 1.4 and as such can now not use FastAPI Users as it has a hard dependency on 1.3.

**Comment by frankie567** [comment]
Sure! I try to keep up with new versions as fast as I can 😄

**Reply by frankie567** [reply]
Unfortunately, we'll have to wait a bit since `encode/databases` is not ready yet for SQLAlchemy 1.4: https://github.com/encode/databases/issues/298

**Reply by silllli** [reply]
Is `encode/databases` actually still needed if there is async support in SQLAlchemy 1.4?

**Reply by Timusan** [reply]
In theory: Yes. But SQLAlchemy's team states that in the 1.4 release (which is an API preparation to 2.0) we should consider the async support to shift around between releases and thus be considered "alpha": https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#asynchronous-io-support-for-core-and-orm.

**Reply by Timusan** [reply]
SQLAlchemy's own async support is now considered Beta with a statement that it's now "unlikely for there to be significant backwards-incompatible changes". So @silllli 's statement is maybe up for consideration again?

More info: https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html?highlight=async

**Reply by AntonOfTheWoods** [reply]
@frankie567 while there are a lot of hugely popular `encode` projects, there are also projects that get relied upon but completely abandoned by them (`broadcaster`, etc.). Absolutely no responses on the repos, they just totally lose interest and require forks and recreated pypis to fix bugs (probably unless a client/you contract with their consulting firm). It looks like support for `sqlalchemy` >= 1.4 will suffer the same fate - after all, what is the point of `databases` if `sqlalchemy` properly supports `async`?
As the async functionality is pretty stable in `sqlalchemy`, might you consider PRs to remove `databases` in favour of native `sqlalchemy`?

**Reply by frankie567** [reply]
@AntonOfTheWoods I don't want to presume about the plans of Encode regarding those projects. Currently, `encode/databases` is still the way described in FastAPI documentation: https://fastapi.tiangolo.com/advanced/async-sql-databases/

Nevertheless, I would be open to review a new database adapter for SQLAlchemy 1.4; but I wish to keep the current one for the moment.
